addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -283,6 +283,7 @@ def _import_invoice_ubl_cii(self, invoice, file_data, new=False):
 
         # Update the invoice.
         invoice.move_type = move_type
+        invoice.journal_id = journal
```

Comment:
Doesn't really make sense with the check above?!
```python
journal = invoice.journal_id
invoice.journal_id = journal
```

Reply:
Actually this is done because when we enter the method `_import_invoice_ubl_cii`, the invoice has already a journal (which was set here
https://github.com/odoo/odoo/blob/saas-16.2/addons/account/models/account_journal.py#L746-L751) which is overriden by `invoice.move_type = move_type` because of https://github.com/odoo/odoo/blob/saas-16.2/addons/account/models/account_move.py#L664-L671
So that's why I have to set it back again to the previous value

Reply:
The compute function should not change the journal if it is consistent with the move type

Reply:
Maybe we can even change the dependence on `move_type` in the compute of `journal_id`

Reply:
> It will

But it shouldn't; you can fix that :D

Reply:
This line fixes it for all cases ^^

If we do this:
```py
self.env['account.journal'].with_context(default_journal_id=journal3.id)._create_document_from_attachment(attachment.id)
```
Then `_search_default_journal` is called and it replaces the current journal which is not what we want. So I guess we could simply fetch the journal from context in `_search_default_journal` and set it from there?

But nope, because if we do this:
```py
new_invoice = journal2._create_document_from_attachment(attachment.id)
```
Then `_search_default_journal` is called too, but this time the journal is replaced too (because `journal2` is not in the context) which is not what we want.

Something that would work to fix both of theses problems would be to skip the moves which already have a journal and avoid overriding `journal2` form earlier.
```py
    @api.depends('move_type')
    def _compute_journal_id(self):
        for record in self.filtered(lambda r: not r.journal_id):
            record.journal_id = record._search_default_journal()
```
However this would break other flows such as:
- Create an invoice
- Use the "Switch to invoice/credit note" button
- The move type is changed to refund, but the journal is not updated because of this filtering...

So to me this fix is the one that's seems the easiest to fix all these (we could also use a key context to not override the journal is some cases but that would be a worse solution IMO)